average.js

```diff
@@ -1,6 +1,13 @@
 
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  let realLength = numbers.length;
+  return (
+    numbers.reduce((p, c) => {
+      if (c) return p + c;
+      realLength--;
+      return p;
+    }, 0) / realLength
+  );
 }
 
 module.exports = {average};
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
Consider this situation: IoT sensors sometimes report a lot of variation in their readings. If you consider that the sensor is reading out battery temperature, there cannot be sudden variation in reality. Hence, a lot of variation in the readings would mean a defective sensor.
Yet, our average function would compute the average as though everything was ok.
Add functionality: Come up with a way to return NaN from the average function, if the variation is too much.
You will need to:

- determine the meaning of 'too much'. E.g., the difference between max and min, or the difference in successive readings. Pick the method you feel is best. Include any references you used for it.
- implement a test to check that the average function returns NaN on high variance. Name the test appropriately.
- implement the functionality to pass the test.

Reply:
@numocityadmin  Please check the recent push, all the test cases are passing but there is build error
![image](https://user-images.githubusercontent.com/68724228/146205513-62b13f8d-a5d9-4e8a-9328-78c76e1c8847.png)
Here is my code for average.js
```
/**
 * @param {array} numbers ;
 * @param {number} maxVariance  difference between max and min : default 2;
 * @param {number} maxEachVariance value of high variance in successive /
 *  reading default 10;
 * @param {number} highVarianceOccurrence no of occurrence of high variance /
 *  in successive readings : default 0 ;
 * @returns average of the giver array if no violation else Nan ;
 */

function average(numbers, maxVariance = 0, maxEachVariance = 0,
    highVarianceOccurrence = 0) {
  let realLength = numbers.length;
  const avg = numbers.reduce((p, c) => {
    if (!isNaN(c)) return p + c;
    realLength--;
    return p;
  }, 0) / realLength;

  let isExtremeVariance = false;
  const nonNanNumbers = numbers.filter((number) => !isNaN(number));
  // For maximum variance
  if (maxVariance) {
    const min = Math.min(...nonNanNumbers);
    const max = Math.max(...nonNanNumbers);
    (max-min) > maxVariance ? isExtremeVariance = true : null;
  }
  // For each variance
  if (maxEachVariance) {
    for (let i = 0; i < nonNanNumbers.length-1; i++) {
      if (Math.abs(nonNanNumbers[i]-nonNanNumbers[i+1]) > maxEachVariance) {
        highVarianceOccurrence--;
      }
    }
    highVarianceOccurrence < 0 ? isExtremeVariance = true : null;
  }
  // return the average
  return !isExtremeVariance ? avg : NaN;
}

module.exports = {average};
````
and test case result:
![image](https://user-images.githubusercontent.com/68724228/146205756-45008eb5-d54a-46c7-973e-decae5ff911f.png)